Twilio/AccessToken.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+include_once 'JWT.php';
+include_once 'Auth/ConversationsGrant.php';
+include_once 'Auth/IpMessagingGrant.php';
+include_once 'Auth/SyncGrant.php';
+
+class Twilio_AccessToken
+{
+    private $signingKeySid;
+    private $accountSid;
+    private $secret;
+    private $ttl;
+    private $identity;
+    private $nbf;
+    private $grants;
+
+    public function __construct($accountSid, $signingKeySid, $secret, $ttl = 3600, $identity = null)
+    {
+        $this->signingKeySid = $signingKeySid;
+        $this->accountSid = $accountSid;
+        $this->secret = $secret;
+        $this->ttl = $ttl;
+
+        if (!is_null($identity)) {
+            $this->identity = $identity;
+        }
+
+        $this->grants = array();
+    }
+
+    /**
+     * Set the identity of this access token
+     *
+     * @param string $identity identity of the grant
+     *
+     * @return Twilio_AccessToken updated access token
+     */
+    public function setIdentity($identity)
+    {
+        $this->identity = $identity;
+        return $this;
+    }
+
+    /**
+     * Returns the identity of the grant
+     *
+     * @return string the identity
+     */
+    public function getIdentity()
+    {
+        return $this->identity;
+    }
+
+    /**
+     * Set the nbf of this access token
+     *
+     * @param integer $nbf nbf in epoch seconds of the grant
+     *
+     * @return Twilio_AccessToken updated access token
+     */
+    public function setNbf($nbf)
+    {
+        $this->nbf = $nbf;
+        return $this;
+    }
+
+    /**
+     * Returns the nbf of the grant
+     *
+     * @return integer the nbf in epoch seconds
+     */
+    public function getNbf()
+    {
+        return $this->nbf;
+    }
+
+    /**
+     * Add a grant to the access token
+     *
+     * @param Twilio_Auth_Grant $grant to be added
+     *
+     * @return $this the updated access token
+     */
+    public function addGrant(Twilio_Auth_Grant $grant)
+    {
+        $this->grants[] = $grant;
+        return $this;
+    }
+
+
+    public function toJWT($algorithm = 'HS256')
+    {
+        $header = array(
+            'cty' => 'twilio-fpa;v=1',
+            'typ' => 'JWT'
+        );
+
+        $now = time();
+
+        $grants = array();
+        if ($this->identity) {
```

Comment:
Note: This condition is incorrect, needs to be `if (!is_null($this->identity))`.